File: frontend/replace_post.py
import os
import re

replacements = [
    # Exact capitalized phrases
    (r'\bCreate Post\b', 'Create Sela'),
    (r'\bBoost Post\b', 'Boost Sela'),
    (r'\bDelete Post\b', 'Delete Sela'),
    (r'\bMute this post\b', 'Mute this sela'),
    (r'\bUnmute this post\b', 'Unmute this sela'),
    (r'\bDeleting post\.\.\.', 'Deleting sela...'),
    (r'\bFailed to delete post\b', 'Failed to delete sela'),
    (r'\bPost Impressions\b', 'Sela Impressions'),
    (r'\bPost media\b', 'Sela media'),
    (r'\bPost your reply\b', 'Drop your reply'), # Custom adjustment for "Drop your reply" or "Sela your reply"? The user said "Change all text Post to Sela". So "Sela your reply"? "Sela" can be a verb. I'll use "Drop your reply" because they had "Drop your thought". Wait, they said change "Post" to "Sela". I'll use "Sela your reply".
    (r'\bPost your reply\b', 'Sela your reply'),
    
    # Text inside JSX tags
    (r'>Post<', '>Sela<'),
    (r'>Posts<', '>Selas<'),
    (r'>Posts\b', '>Selas'),
    (r'>\s*Post\s*<', '>Sela<'),
    
    # Specific sentences
    (r'liked your post', 'liked your sela'),
    (r'resela\'d your post', 'resela\'d your sela'),
    (r'replied to your post', 'replied to your sela'),
    (r'quoted your post', 'quoted your sela'),
    (r'Check out this post', 'Check out this sela'),
    (r'No posts in this space', 'No selas in this space'),
    (r'pending posts', 'pending selas'),
    
    # Generic lowercase with "this" or "a"
    (r'\bthis post\b', 'this sela'),
    (r'\ba post\b', 'a sela'),
    (r'\byour post\b', 'your sela'),
    
    # Plurals
    (r'12\.5K posts', '12.5K selas'),
    (r'8,432 posts', '8,432 selas'),
    (r'5,210 posts', '5,210 selas'),
    (r'3,100 posts', '3,100 selas'),
    (r'\bK posts\b', 'K selas'),
    
    # Buttons/Actions
    (r'"Post"', '"Sela"'),
    (r"'Post'", "'Sela'"),
]

def process_directory(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.tsx', '.ts')):
                filepath = os.path.join(root, file)
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                new_content = content
                for pattern, replacement in replacements:
                    new_content = re.sub(pattern, replacement, new_content)
                
                if new_content != content:
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Updated {filepath}")

process_directory('c:/Users/HP/Desktop/intasela/frontend/src')
process_directory('c:/Users/HP/Desktop/intasela/admin-frontend/src')

File: frontend/replace_post.js
const fs = require('fs');
const path = require('path');

const replacements = [
    // Exact capitalized phrases
    [/\bCreate Post\b/g, 'Create Sela'],
    [/\bBoost Post\b/g, 'Boost Sela'],
    [/\bDelete Post\b/g, 'Delete Sela'],
    [/\bMute this post\b/g, 'Mute this sela'],
    [/\bUnmute this post\b/g, 'Unmute this sela'],
    [/\bDeleting post\.\.\./g, 'Deleting sela...'],
    [/\bFailed to delete post\b/g, 'Failed to delete sela'],
    [/\bPost Impressions\b/g, 'Sela Impressions'],
    [/\bPost media\b/g, 'Sela media'],
    [/\bPost your reply\b/g, 'Drop your reply'],
    
    // Text inside JSX tags
    [/>Post</g, '>Sela<'],
    [/>Posts</g, '>Selas<'],
    [/>Posts\b/g, '>Selas'],
    [/>\s*Post\s*</g, '>Sela<'],
    
    // Specific sentences
    [/liked your post/g, 'liked your sela'],
    [/resela\'d your post/g, "resela'd your sela"],
    [/replied to your post/g, 'replied to your sela'],
    [/quoted your post/g, 'quoted your sela'],
    [/Check out this post/g, 'Check out this sela'],
    [/No posts in this space/g, 'No selas in this space'],
    [/pending posts/g, 'pending selas'],
    
    // Generic lowercase with "this" or "a"
    [/\bthis post\b/g, 'this sela'],
    [/\ba post\b/g, 'a sela'],
    [/\byour post\b/g, 'your sela'],
    
    // Plurals
    [/12\.5K posts/g, '12.5K selas'],
    [/8,432 posts/g, '8,432 selas'],
    [/5,210 posts/g, '5,210 selas'],
    [/3,100 posts/g, '3,100 selas'],
    [/\bK posts\b/g, 'K selas'],
    
    // Buttons/Actions
    [/"Post"/g, '"Sela"'],
    [/'Post'/g, "'Sela'"],
];

function processDirectory(directory) {
    const files = fs.readdirSync(directory);
    
    for (const file of files) {
        const fullPath = path.join(directory, file);
        if (fs.statSync(fullPath).isDirectory()) {
            processDirectory(fullPath);
        } else if (fullPath.endsWith('.tsx') || fullPath.endsWith('.ts')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            let newContent = content;
            
            for (const [pattern, replacement] of replacements) {
                newContent = newContent.replace(pattern, replacement);
            }
            
            if (newContent !== content) {
                fs.writeFileSync(fullPath, newContent, 'utf8');
                console.log(`Updated ${fullPath}`);
            }
        }
    }
}

processDirectory(path.join(__dirname, 'src'));
processDirectory(path.join(__dirname, '../admin-frontend/src'));
